pkg/kubectl/apply/element.go

```diff
@@ -325,6 +326,40 @@ func IsAdd(e Element) bool {
 	return e.HasLocal() && !e.HasRemote()
 }
 
+// IsConflict returns true if the field represented by e should both have the recorded value
+// and remote value, but the values are not same
+func IsConflict(e Element) bool {
+	if e.HasRecorded() && e.HasRemote() {
+		var record map[string]interface{}
+		var remote map[string]interface{}
+		switch e.(type) {
+		case ListElement:
+			return !reflect.DeepEqual(e.(ListElement).ListElementData.GetRecordedList(), e.(ListElement).ListElementData.GetLocalList())
```

Comment:
Will this work?  What if the lists have elements with lists that need to be compared?

Reply:
Actually I wanted to have a comment here to discuss my concerns about conflict definition in ListElement. It is very different to maps, in which conflict is definite as long as same keys have different values. ListElement objects holds remote, recorded fields in form of []interface. The problem can be converted to "compare two []interface "
Also I noticed your comments in https://github.com/kubernetes/kubernetes/blob/cdecea545553eff09e280d389a3aef69e2f32bf1/pkg/kubectl/apply/strategy/merge.go#L23-L27

I want talk about some corner cases in ListElement:
1. Does it involve `literal equality` or `deep equality` ? 
case 1:
```
record:  ["message1", "message2", "message3"]
remote:  ["message1", "message2"]
Because different number of elements, so record and remote are "not equal" . Is it a "conflict" ?
```
2. Does it require`nested validation` ? 
case 2
```
record: [{"Ports": [8080, 443]} ]
remote: [{"Ports": [8080]}]
They both hold struct as element, in which values are different (remote is missing 443), is it a conflict ?
```

My PR 01fb875 is only validating if the two ListElement object are "deep equality", for example, case 1 and case 2 will be catch as 'conflict'

I think the `conflict` may be not only about objects are equality, 'conflicts' also considers if the elements of the two list (remote and record) can be merged. In this case, case1 and case2 could be `not conflict`, is it correct ?